### ISSUE TITLE
Desperate: Vladivostok (appid 1832070)

### ISSUE THREAD

**Issue by Patola** [ValveSoftware/Proton#6533]
# Compatibility Report
- Name of the game with compatibility issues: Desperate: Vladivostok
- Steam AppID of the game: 1832070

## System Information
- GPU: RX 6800 XT
- Driver/LLVM version: mesa 22.3.4-1
- Kernel version: 6.0.19-273-tkg-bore
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/Patola/d15efddcc566783e100b1726c9824228
- Proton version: 7.0-6

## I confirm:
- [X] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.

## Symptoms 
- Game is VR only, it doesn't start. When launching, an unreal engine crash screen quickly appears and disappears on the screen when launching it. Using Valve Index as VR

## Reproduction
- Just try and run the game. Be it from the green PLAY button on Steam or via SteamVR Home. Same thing.
[steam-1832070.log](https://github.com/ValveSoftware/Proton/files/10727675/steam-1832070.log)

**Comment by kisak-valve** [comment]
Hello @Patola, `err:combase:RoGetActivationFactory Failed to find library for L"Windows.Graphics.Holographic.HolographicSpace"` looks like a line of interest followed by an access violation (c0000005) in your log.

**Comment by mohamadaljaf** [comment]
Hi, I added the interface for HolographicSpace and implemented some functions that should make the game work. Can you please test it again? I'm not sure if my commits were pulled into Proton, but they were merged in Wine: https://source.winehq.org/git/wine.git/commitdiff/18453129be33fd50d3050dec722df054a181017a

**Comment by Patola** [comment]
That's amazing to know! I will prepare a proton build with your patch, test the games and get back to you. Thanks!

**Comment by Patola** [comment]
Trying to merge your wine changes to proton, but currently having issues because it seemingly needs a newer version of `mingw-headers` which has `ivectorchangedeventargs.h`. Still trying

**Comment by Patola** [comment]
Yeah, I don't think I will be able to compile the current proton with your patch, @mohamadaljaf. You modify an already existing code from idl that is in wine 8.6 but not wine-proton 7.0. And there is some change in types between these two versions, and then when I try to change this there is a cascade of other issues. I guess I will have to wait for Valve to resynch their proton fork... As of now I can't see any 8.x branch.

**Comment by mohamadaljaf** [comment]
I see, thanks for letting me know, that's unfortunate. Are there any custom builds of proton that are based on Wine 8.x? If not, I guess the only thing left to do is wait for Proton 8.0 to come out.

**Comment by Patola** [comment]
Not that I know of. Even Proton-GE hasn't gone 8.x yet and Proton-TKG is no more.

**Comment by Patola** [comment]
We just have the first proton build using wine 8... https://github.com/ValveSoftware/Proton/tree/proton-8.0-1c
Will try with that

**Comment by Patola** [comment]
Unsuccessful, tried a lot but I've got limited time for this and my life, work, etc...
The new Proton-GE is out, even GloriousEggroll did not base it on the newest wine but Proton 8.0-1c, which doesn't have the new UWP libraries for the game to run.

But nevertheless even if I was successful, now SteamVR is completely broken on my arch linux and doesn't even turn on the headset (on windows it works fine). So I can't really test any build right now.

**Comment by Patola** [comment]
@mohamadaljaf it's two years later and it seems this interface was still not implemented. I saw even on Proton-GE that they tried it then removed this library.

**Comment by Patola** [comment]
So we're on Proton 9 now but there is still another game (Pirates VR: Jolly Roger) affected by that? I don't understand, it should have been part of proton already, no? It's two years later!

I've tried running Desperate: Vladivostok again with the latest proton 9 and it still shows that Unreal Engine screen.

@mohamadaljaf since you added the library, do you know what might be happening?

**Comment by mohamadaljaf** [comment]
Could you attach the Proton 9 log?

**Comment by Patola** [comment]
Sure, here you go. 

[steam-1832070.log](https://github.com/user-attachments/files/18814616/steam-1832070.log)

**Comment by Patola** [comment]
@kisak-valve any chance of using Proton 10.0 to try and add complete holographicscape support?

**Comment by Patola** [comment]
@mohamadaljaf used your patch given in #8457 to re-test this one too and got the same kind of unreal engine error dialog. Pasting the proton log here if that would be useful to you.

Note: running with SteamVR via Steam Link

[steam-1832070.log](https://github.com/user-attachments/files/23288344/steam-1832070.log)

**Comment by Patola** [comment]
This bug was fixed by @mohamadaljaf with his three patches to the proton_10.0 branch. I have created a proton build to run these games while it is not integrated in mainline proton. This is the build: https://github.com/Patola/wine/releases/tag/protola-10.0

The fix came from #8457
